cmd/controller/main.go

```diff
@@ -49,34 +53,36 @@ func init() {
 
 // Options for running this binary
 type Options struct {
-	EnableVerboseLogging bool
-	MetricsPort          int
-	HealthProbePort      int
+	MetricsPort     int
+	HealthProbePort int
 }
 
 func main() {
-	flag.BoolVar(&options.EnableVerboseLogging, "verbose", false, "Enable verbose logging")
 	flag.IntVar(&options.MetricsPort, "metrics-port", 8080, "The port the metric endpoint binds to for operating metrics about the controller itself")
 	flag.IntVar(&options.HealthProbePort, "health-probe-port", 8081, "The port the health probe endpoint binds to for reporting controller health")
 	flag.Parse()
 
-	log.Setup(
-		controllerruntimezap.UseDevMode(options.EnableVerboseLogging),
-		controllerruntimezap.ConsoleEncoder(),
-		controllerruntimezap.StacktraceLevel(zapcore.DPanicLevel),
-	)
-	manager := controllers.NewManagerOrDie(controllerruntime.GetConfigOrDie(), controllerruntime.Options{
+	config := controllerruntime.GetConfigOrDie()
+	clientSet := kubernetes.NewForConfigOrDie(config)
+
+	// Setup knative/pkg style logging
+	ctx, startinformers := injection.EnableInjectionOrDie(signals.NewContext(), config)
+	go startinformers()
+	logger, atomicLevel := sharedmain.SetupLoggerOrDie(ctx, component)
+	ctx = logging.WithLogger(ctx, logger)
+	cmw := informer.NewInformedWatcher(clientSet, system.Namespace())
+	cmw.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
```

Comment:
This code block looks honestly pretty confusing at first, and clutters up the main() function. I see the comment, but is it possible to separate this out into a helper function that summarizes what is happening here, like `SetUpLogging()`? 

I'd love to understand what is being used to set up logging in Karpenter without having to dive deep into another package.